src/classes/CRLP_RecalculateBTN_TEST.cls

```diff
@@ -172,4 +172,367 @@ private class CRLP_RecalculateBTN_TEST {
         Test.stopTest();
     }
 
+    /*********************************************************************************************************
+    * @description Given Household Setting → Always Rollup To Primary Contact = FALSE, and a Household Contact
+    * donates a gift, and his organization matches the gift, when the Recalculate Rollup logic runs,
+    * Soft Credit (Amount equals to Organization match) should be given to the Contact.
+    */
+    @isTest static void test_CalculateContactSoftCredit_AlwaysRollupToPrimaryContactIsFalse() {
+        String role = 'Soft Credit';
+        UTIL_CustomSettingsFacade.getHouseholdsSettingsForTests(new npo02__Households_Settings__c (
+            npo02__Always_Rollup_to_Primary_Contact__c = false,
+            npo02__Soft_Credit_Roles__c = role
+        ));
+        deactivateTriggers();
+
+        Account household = new Account(
```

Comment:
The unit test data set (and most of the logic) looks very very similar for these 4 new test methods. Is it possible to combine these into a single shared method that combines logic where needed. See the `CRLP_RollupContact_TEST` class as an example of using a single test method for 6 different tests.

Reply:
Also, this test class should be focused on testing that the button controller works. Tests that validate the actual rollup logic should be in the unit test classes that focus on rollups. That might be `CRLP_RollupContact_TEST` for Customizable Rollups or `RLLP_OppPartialSoftCreditRollup_TEST` for legacy rollups. I also think the `CRLP_RollupContact_TEST` class is already testing the 'AlwaysRollupToPrimary` logic.

Reply:
For this scenario, I think combining the test scenarios into a single method can make it complicated to read. But you have a good point, I can move the creation of the Test data into separate methods that can be called by the test methods.

Reply:
Thanks. The tests related to the AlwaysRollupToPrimary option should be moved as well since they're not related to the button.